// figma-js/build/module/index.js
var __assign = (this && this.__assign) || function () {
    __assign = Object.assign || function(t) {
        for (var s, i = 1, n = arguments.length; i < n; i++) {
            s = arguments[i];
            for (var p in s) if (Object.prototype.hasOwnProperty.call(s, p))
                t[p] = s[p];
        }
        return t;
    };
    return __assign.apply(this, arguments);
};
export * from './figmaTypes';
import axios from 'axios';
export var Client = function (opts) {
    var headers = opts.accessToken
        ? {
            Authorization: "Bearer " + opts.accessToken,
        }
        : {
            'X-Figma-Token': opts.personalAccessToken,
        };
    var client = axios.create({
        baseURL: "https://" + (opts.apiRoot || 'api.figma.com') + "/v1/",
        headers: headers,
    });
    return {
        client: client,
        file: function (fileId, params) {
            if (params === void 0) { params = {}; }
            return client.get("files/" + fileId, {
                params: __assign(__assign({}, params), { ids: params.ids ? params.ids.join(',') : '' }),
            });
        },
        fileVersions: function (fileId) { return client.get("files/" + fileId + "/versions"); },
        fileNodes: function (fileId, params) {
            return client.get("files/" + fileId + "/nodes", {
                params: __assign(__assign({}, params), { ids: params.ids.join(',') }),
            });
        },
        fileImages: function (fileId, params) {
            return client.get("images/" + fileId, {
                params: __assign(__assign({}, params), { ids: params.ids.join(',') }),
            });
        },
        fileImageFills: function (fileId) { return client.get("files/" + fileId + "/images"); },
        comments: function (fileId) { return client.get("files/" + fileId + "/comments"); },
        postComment: function (fileId, params) {
            return client.post("files/" + fileId + "/comments", params);
        },
        deleteComment: function (fileId, commentId) {
            return client.delete("files/" + fileId + "/comments/" + commentId);
        },
        me: function () { return client.get("me"); },
        teamProjects: function (teamId) { return client.get("teams/" + teamId + "/projects"); },
        projectFiles: function (projectId) { return client.get("projects/" + projectId + "/files"); },
        teamComponents: function (teamId, params) {
            if (params === void 0) { params = {}; }
            return client.get("teams/" + teamId + "/components", { params: params });
        },
        fileComponents: function (fileId) { return client.get("files/" + fileId + "/components"); },
        component: function (key) { return client.get("components/" + key); },
        teamStyles: function (teamId, params) {
            if (params === void 0) { params = {}; }
            return client.get("teams/" + teamId + "/styles", { params: params });
        },
        fileStyles: function (fileId) { return client.get("files/" + fileId + "/styles"); },
        style: function (key) { return client.get("styles/" + key); },
    };
};
